### PR TITLE
XW-4743 | Fandom App webview events tracking added.

### PR DESCRIPTION
**Pull request by kolek7**
## Links

https://wikia-inc.atlassian.net/browse/XW-4743

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: app/utils/track.js [hunk 1]
```diff
@@ -120,9 +120,12 @@ function isPageView(category) {
  * @returns {void}
  */
 export function track(params) {
+	const isFandomApp = new URL(window.location).searchParams.get('mobile-app');
+	const appOrMercury = (isFandomApp === 'true' ? 'fandom-app-' : 'mercury-');
```
Comment (rybmat) [inline app/utils/track.js]: can we name this const differently? `categoryPrefix` for example?

File: app/utils/track.js [hunk 2]
```diff
@@ -120,9 +120,12 @@ function isPageView(category) {
...
+
 	const trackingMethod = params.trackingMethod || 'both',
 		action = params.action,
-		category = params.category ? `mercury-${params.category}` : null,
+		category = params.category ? appOrMercury + params.category : null,
```
Comment (rybmat) [inline app/utils/track.js]: we can use string templates here `${categoryPrefix}-${params.category}`
(note: remove trailing `-` from the categoryPrefix const)

**Review by rybmat — CHANGES_REQUESTED** [review]

**Review by hakubo — APPROVED** [review]

File: app/utils/track.js [hunk 3]
```diff
@@ -120,9 +120,12 @@ function isPageView(category) {
  * @returns {void}
  */
 export function track(params) {
+	const isFandomApp = new URL(window.location).searchParams.get('mobile-app') === 'true';
+	const trackingCategoryPrefix = (isFandomApp ? 'fandom-app' : 'mercury');
+
 	const trackingMethod = params.trackingMethod || 'both',
```
Comment (hakubo) [inline app/utils/track.js]: while you're at it could you split to const per variable?

File: app/utils/track.js [hunk 4]
```diff
@@ -120,9 +120,12 @@ function isPageView(category) {
  * @returns {void}
  */
 export function track(params) {
+	const isFandomApp = new URL(window.location).searchParams.get('mobile-app') === 'true';
```
Comment (hakubo) [inline app/utils/track.js]: this function runs also on Fastboot
on  line 133 and 137 there are neede checks - this should probably happen after these lines

Comment (xkxd) [reply app/utils/track.js]: i'd suggest moving those fastboot/window checks at the top of `track` method

**Review by rybmat — APPROVED** [review]

**Review by hakubo — APPROVED** [review]

**Pull request merged**